### PR TITLE
PR: Allow for multiple installs and update buttons to reflect state

### PR DESCRIPTION
**Pull request by goanpeca**
# Description

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# References

- This now allows to install and uninstall multiple packages at the same time.
- The final refresh will only happen when all the queued processes have finished.
- Updates text and color when clicking on buttons.
- Removes showind the output by default (this is an advanced user feature)

# How has this been tested?

- [x] example: the test suite for my feature covers cases x, y, and z
- [x] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

napari

### CODE REVIEW

**Review by tlambert03 — APPROVED** [review]
LGTM 🚀

**Review by sofroniewn — APPROVED** [review]
LGTM!

File: napari/_qt/dialogs/qt_plugin_dialog.py [hunk 1]
```diff
@@ -80,26 +101,63 @@ def __init__(
...
+        process.finished.connect(
+            lambda ec, es: self._on_process_finished(process, ec, es)
+        )
+        return process
 
     def set_output_widget(self, output_widget: QTextEdit):
         if output_widget:
             self._output_widget = output_widget
-            self.process.setParent(output_widget)
+            # self.process.setParent(output_widget)
```
Comment (jni) [inline napari/_qt/dialogs/qt_plugin_dialog.py]: do you want to either remove this or keep it? =)

File: napari/_qt/dialogs/qt_plugin_dialog.py [hunk 2]
```diff
@@ -353,14 +434,31 @@ def addItem(
             plugin_name=plugin_name,
             enabled=enabled,
         )
-        method = getattr(
-            self.installer, 'uninstall' if plugin_name else 'install'
+        item.widget = widg
+        # method = getattr(
+        #     self.installer, 'uninstall' if plugin_name else 'install'
+        # )
```
Comment (jni) [inline napari/_qt/dialogs/qt_plugin_dialog.py]: As above — remove comments or uncomment, not sure here. =)

**Review by jni — APPROVED** [review]
Just found a couple of straggling comments 🤷 but otherwise very happy with this functionality!

File: napari/_qt/dialogs/qt_plugin_dialog.py (hunk 1 above)
Comment (goanpeca) [reply napari/_qt/dialogs/qt_plugin_dialog.py]: Fixed!

File: napari/_qt/dialogs/qt_plugin_dialog.py (hunk 2 above)
Comment (goanpeca) [reply napari/_qt/dialogs/qt_plugin_dialog.py]: Fixed!

**Pull request merged**